make_prg/from_msa/prg_builder.py

```diff
@@ -103,94 +109,128 @@ def get_sub_alignment_by_list_id(
+        variant_prgs = []
+        num_clusters = len(clustered_ids)
+
+        list_sub_alignments = [
+            self.get_sub_alignment_by_list_id(
+                id_list, self.alignment, [interval.start, interval.stop]
+            )
+            for id_list in clustered_ids
+        ]
+
+        if interval.start not in self.subAlignedSeqs:
+            self.subAlignedSeqs[interval.start] = []
+        else:
+            logging.debug(
+                "subAlignedSeqs already had key %d in keys: %s. This shouldn't happen.",
+                interval.start,
+                list(self.subAlignedSeqs.keys()),
+            )
+
+        while len(list_sub_alignments) > 0:
+            sub_alignment = list_sub_alignments.pop(0)
+            sub_aligned_seq = PrgBuilder(
+                msa_file=self.msa_file,
+                alignment_format=self.alignment_format,
+                max_nesting=self.max_nesting,
+                nesting_level=self.nesting_level + 1,
+                min_match_length=self.min_match_length,
+                site=self.site,
+                alignment=sub_alignment,
+                interval=interval,
+            )
+            variant_prgs.append(sub_aligned_seq.prg)
+            self.site = sub_aligned_seq.site
+            self.subAlignedSeqs[interval.start].append(sub_aligned_seq)
+
+        assert num_clusters == len(variant_prgs), (
+            "I don't seem to have a sub-prg sequence for all parts of the partition - there are %d "
+            "classes in partition, and %d variant seqs"
+            % (num_clusters, len(variant_prgs))
+        )
+        return variant_prgs
+
+    def get_variants(self, interval) -> Sequences:
+        variant_prgs = []
+        if self.skip_clustering(
+            interval,
+            self.nesting_level,
+            self.max_nesting,
+            self.min_match_length,
+            self.alignment,
+        ):
+            sub_alignment = self.alignment[:, interval.start : interval.stop + 1]
+            variant_prgs = get_expanded_sequences(sub_alignment)
+            logging.debug(f"Variant seqs found: {variant_prgs}")
+        else:
+            clustering_result = kmeans_cluster_seqs_in_interval(
+                [interval.start, interval.stop],
+                self.alignment,
+                self.min_match_length,
+            )
+            if clustering_result.no_clustering:
+                logging.debug(
+                    "Clustering did not group any sequences together, each seq is a cluster"
+                )
+                variant_prgs = clustering_result.sequences
+                logging.debug(f"Variant seqs found: {variant_prgs}")
+            else:
+                variant_prgs = self.prg_recur(interval, clustering_result.clustered_ids)
+        assert len(variant_prgs) > 1, "Only have one variant seq"
+        assert len(variant_prgs) == len(
+            list(remove_duplicates(variant_prgs))
+        ), "have repeat variant seqs"
```

Comment:
Are we able to suggest a solution here? Having run into some of these assert statements in the past I can attest that they are very confusing and I generally have no idea what to do next